runtime/bcutil/ClassFileOracle.hpp

```diff
@@ -921,6 +923,16 @@ class NameAndTypeIterator
 	bool annotationRefersDoubleSlotEntry() const { return _annotationRefersDoubleSlotEntry; }
 	bool isInnerClass() const { return _isInnerClass; }
 
+	U_8 constantDynamicType(U_16 cpIndex) const
+	{
+		J9CfrConstantPoolInfo* nas = &_classFile->constantPool[_classFile->constantPool[cpIndex].slot2];
+		if ('D' == nas->bytes[nas->slot1 - 1]) {
+			return JBldc2dw;
+		} else {
+			return JBldc2lw;
+		}
```

Comment:
What about ConstantDynamics that don't return long or double?

Reply:
this is a helper for `ldc2_w` bytecode marking which requires a special bytecode entry (jit optimization)
For ConstantDynamics that are `ldc` or `ldc_w` they will not require a special bytecode entry
(ie. `addBytecodeFixupEntry(entry++, codeIndex + 1, cpIndex, ConstantPoolMap::REFERENCED);` should cover that case)

Reply:
I need more detail on this.  Not currently following.

Reply:
The VM have special bytecode (`JBldc2dw / JBldc2lw`) for loading with long and double in the interpreter depending wether it is a long or double (I believe it was for jit optimization).

this function is used to help determine the expected return type from `ldc2_w` instruction and return the correct fixup bytecode entry.

In the case that ConstantDynamic used in `ldc / ldc_w`, there is no special bytecode for different types, so only a fixup entry is required.
ie. `ClassFileOracle.cpp - line 1455 && 1506 > CFR_BC_ldc and CFR_BC_ldc_w`